framework/wazuh/syscheck.py

```diff
@@ -81,20 +81,15 @@ def clear(agent_list=None):
     result = AffectedItemsWazuhResult(all_msg='Syscheck database was cleared on returned agents',
                                       some_msg='Syscheck database was not cleared on some agents',
                                       none_msg="No syscheck database was cleared")
-    wdb_conn = WazuhDBConnection()
 
     system_agents = get_agents_info()
+    wdb_conn = WazuhDBConnection()
```

Comment:
Is there any scenario where agent_list can be an empty list at this point? In such case, we should not open a socket connection that is not going to be used.

Also, please ensure the object is properly collected, destroyed and the connection closed when the clear function fishishes.

Reply:
`wdb_conn` could be initialized without need when requesting the endpoint with a wrong agent ID. I added a little fix to only initialize it when it's totally necessary.

As for the second request I've manually closed the connection to the db in my last commits to this branch.